tests/test_sources.py

```diff
@@ -13,6 +13,7 @@
 def fancy_wheel(tmp_path):
     path = tmp_path / "fancy-1.0.0-py2.py3-none-any.whl"
     files = {
+        "fancy/": b"""""",
```

Comment:
If feel these directory entries should use `None` as values instead (and change the check below). Hard-coding the `/` logic in tests does not feel right to me.

Reply:
It seems like the only way to get ZipFile to add directory entries is to write an empty item that ends with "/". Setting it to `"fancy/": None,` breaks when it tries to write the data to the entry.

I do agree that having the test effectively running the same code as the function we are testing is not the best test implementation. It might be that pulling out the functionality of getting only the files from the zip would be a cleaner way of handling it.

Reply:
Quite sad, but I guess we can only use what's available to us.